api/bootstrap/kubeadm/v1beta2/kubeadmconfig_types.go

```diff
@@ -541,8 +542,8 @@ type KubeadmConfig struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// spec is the desired state of KubeadmConfig.
-	// +optional
-	Spec KubeadmConfigSpec `json:"spec,omitempty"`
+	// +required
```

Comment:
I think technically it would be somehow possible to get away with an empty KubeadmConfig, but in reality folks create a KubeadmControlPlane or a KubeadmConfigTemplate, there we default format to cloud-config and that is copied over in the KubeadmConfig.

I think I would prefer not making the spec in KubeadmConfig optional

Reply:
Realized that this one is actually optional, because the CABPK controller adds everything before the KubeadmConfig is actually used (so for users spec is optional)